kong/clustering.lua

```diff
@@ -333,6 +336,66 @@ local function validate_shared_cert()
 end
 
 
+local check_for_revocation_status
+do
+  local get_full_client_certificate_chain = require("resty.kong.tls").get_full_client_certificate_chain
+  check_for_revocation_status = function ()
+    local cert, err = get_full_client_certificate_chain()
+    if not cert then
+      return nil, err
+    end
+
+    local der_cert
+    der_cert, err = ssl.cert_pem_to_der(cert)
+    if not der_cert then
+      return nil, "failed to convert certificate chain from PEM to DER: " .. err
+    end
+
+    local ocsp_url
+    ocsp_url, err = ocsp.get_ocsp_responder_from_der_chain(der_cert)
+    if not ocsp_url then
+      return nil, err or "OCSP responder endpoint can not be determined, " ..
+                         "maybe the client certificate is missing the " ..
+                         "required extensions"
+    end
+
+    local ocsp_req
+    ocsp_req, err = ocsp.create_ocsp_request(der_cert)
+    if not ocsp_req then
+      return nil, "failed to create OCSP request: " .. err
+    end
+
+    local c = http.new()
+    local res
+    res, err = c:request_uri(ocsp_url, {
+      headers = {
+        ["Content-Type"] = "application/ocsp-request"
+      },
+      timeout = OCSP_TIMEOUT,
+      method = "POST",
+      body = ocsp_req,
+    })
+
+    if not res then
+      return nil, "failed sending request to OCSP responder: " .. tostring(err)
+    end
+    if res.status ~= 200 then
+      return nil, "OCSP responder returns bad HTTP status code: " .. res.status
+    end
+
+    local ocsp_resp = res.body
+    if ocsp_resp and #ocsp_resp > 0 then
+      res, err = ocsp.validate_ocsp_response(ocsp_resp, der_cert)
+      if not res then
+        return false, "failed to validate OCSP response: " .. err
+      end
+    end
+
+    return true
```

Comment:
`if not ocsp_resp or #ocsp_resp == 0 `somehow, should we return false instead?

Reply:
Good catch!